frontend/lib/identification/card_detail_view.dart

```diff
@@ -1,55 +1,120 @@
 import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
 import 'package:flutter_svg/svg.dart';
 import 'package:intl/intl.dart';
+import 'package:otp/otp.dart';
+import 'package:provider/provider.dart';
+import 'package:qr_flutter/qr_flutter.dart';
 
+import '../verification/verification_card_details.dart';
+import '../verification/verification_encoder.dart';
 import 'card_details.dart';
+import 'card_details_model.dart';
 import 'id_card.dart';
+import 'otp_generator.dart';
 
 class CardDetailView extends StatelessWidget {
   final CardDetails cardDetails;
   final VoidCallback onOpenQrScanner;
+  final OTPGenerator _otpGenerator;
 
-  const CardDetailView({Key key, this.cardDetails, this.onOpenQrScanner})
-      : super(key: key);
+  CardDetailView({Key key, this.cardDetails, this.onOpenQrScanner})
+      : _otpGenerator = OTPGenerator(
+            cardDetails.base32TotpSecret, 60 * 5, 10, Algorithm.SHA256),
+        super(key: key);
 
   get _formattedExpirationDate =>
       DateFormat('dd.MM.yyyy').format(cardDetails.expirationDate);
 
   @override
   Widget build(BuildContext context) {
-    return Column(
+    var isLandscape = MediaQuery.of(context).size.width >= 500;
```

Comment:
We should avoid hacks like these whenever possible, and I am pretty sure Flutter is advanced enough to provide better solutions.

I can live with not fixing this immediately in favor of merging this huge PR as soon as possible, but we should at least add a TODO so we wont forget.

Reply:
You're right, I will look into it.